pose_estimation:
    camera:
        view_angle:                 [1.1927, 0.7053]

    marker_types:
        left:                       1
        middle:                     2
        right:                      3

    marker_pose:
        marker2:                    [0.4, 0.2]
        marker3:                    [0.8, 0.0]
    
    docking_pose:                   [0.4, 0.1]

kinematics:
    wheel_info: # Wheel orientation is the angle starting from the x-axis.
        wheel_1:
            type:                   FIXED
            origin:                 [0.25, 0.16]
            radius:                 0.05
            wheel_orientation:      1.5707963267948966      # pi / 2
            roller_orientation:     2.356194490192345       # 3*pi / 4
            encoder_ppr:            494
        
        wheel_2:
            type:                   FIXED
            origin:                 [-0.25, 0.16]
            radius:                 0.05
            wheel_orientation:      1.5707963267948966
            roller_orientation:     0.7853981633974483      # pi/4
            revolute_axis:          0
            encoder_ppr:            494
        
        wheel_3:
            type:                   FIXED
            origin:                 [-0.25, -0.16]
            radius:                 0.05
            wheel_orientation:      1.5707963267948966
            roller_orientation:     2.356194490192345       # 3*pi / 4
            revolute_axis:          0
            encoder_ppr:            494
        
        wheel_4:
            type:                   FIXED
            origin:                 [0.25, -0.16]
            radius:                 0.05
            wheel_orientation:      1.5707963267948966
            roller_orientation:     0.7853981633974483      # pi/4
            revolute_axis:          0
            encoder_ppr:            494
    
    solver:
        precision:                  0.0001
        max_trial:                  10
    
    path_generator:
        docking_offset:             0.5

communication:
    arduino:
        dev_keyword:                'ACM'
        id_keyword:                 'ID_SERIAL_SHORT'
        watchdog:   # Old Arduino Uno
            id:                     '55330343434351017072'
            baudrate:               115200
        
        torch:      # New Arduino Uno
            id:                     '75735303831351106241'
            baudrate:               9600
        
        driver:     # Arduino Mega
            id:                     '7543033343735130F041'
            baudrate:               115200
    
    camera:
        dev_keyword:                'video'
        id_keyword:                 'ID_MODEL_ID'
        marker_tracker:
            id:                     '0812'
            img_width:              1280
            img_height:             720
        
        human_tracker:
            id:                     '0810'
            img_width:              640
            img_height:             480